modules/myplots.py

```diff
@@ -0,0 +1,449 @@
+                          np.linspace(0.247, 0.356, n_var), #Min & max values for cylindrical cells [Löbberding]
+                          np.linspace(0.168, 0.528, n_var), #Min & max values for pouch cells [Löbberding]
+                          np.linspace(0.168, 0.528, n_var) #Min & max values for prismatic cells [Löbberding]
+                          )
+                          ],               
+                      "m_cell2system": [{"Cylindrical":cy, "Pouch":po, "Prismatic":pr} 
+                       for cy,po,pr in zip(
+                          np.linspace(max(m_cell2system_min,0.484), 0.672, n_var), #Min & max values for cylindrical cells [Löbberding]
+                          np.linspace(max(m_cell2system_min,0.495), 0.742, n_var), #Min & max values for pouch cells [Löbberding]
+                          np.linspace(max(m_cell2system_min,0.495), 0.742, n_var), #Min & max values for prismatic cells [Löbberding]
+                          )
+                          ],
+                      "c_cell2system": np.linspace(1.94, 2.21, n_var), #Min & max values for 2020 [König] 
+                      "annual_mileage": np.linspace(96_850, 180_000, n_var), # maximum annual mileage
+                      "servicelife": np.arange(2,11), #Payback period considered by large fleets [IEA 2017] until average truck life in Germany [Wolff 2021] 
+                      "c_diesel": np.linspace(1.233/1.19, 2.16/1.19, n_var), #Lowest and highest Diesel cost in Germany between Jan 2021 and March 2022
+                      "c_elec": np.linspace(0.05, 0.44/1.19, n_var), #Value reported by ... et al. and maximum electricity price 
+                      "consumption": [{"crr": crr, "cd_a": cd_a} for crr, cd_a in 
+                                      zip(np.linspace(0.00385, method.vehicle.fr, n_var_cons), #Minimum to average reported coefficient of rolling resistance
+                                          np.linspace(4.325, method.vehicle.cd_a, n_var_cons))] #Minimum to average reported drag area
+                       }
+        
+        selection_variations = {s: variations[s] for s in selection}
+        
+        return selection_variations
+    
+    def param_sensivitiy(self, method, cell, parameters):
+        
+        c_pars = {}
+        payloads = {}
+        volumes = {}
+        con_maxs = []
+        con_refs = []
+        for param, variations in parameters.items():
+            c_pars[param] = []
+            payloads[param] = []
+            volumes[param] = []
+            if param in ["ncycle", "mspec", "Ccharge", "rho_bat"]: #Cell parameters
+                for var in variations:
+                    varcell = cell.copy()
+                    varcell[param] = var
+                    c_par, payload, volume = method.eval_cell(varcell)[0:3]
+                    c_pars[param].append(c_par)
+                    payloads[param].append(payload)
+                    volumes[param].append(volume)   
```

Comment:
You have this block of code multiple times in small variations. You could create a new class for the sensitivity analysis, or use [functional programming](https://www.youtube.com/watch?v=4B24vYj_vaI) to pass custom functions to handle the different cases.